stix_shifter_modules/nozomi/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,549 @@
+import re
+import json
+from os import path
+from datetime import datetime, timedelta
+from stix_shifter_utils.utils import logger
+from stix_shifter_utils.stix_translation.src.patterns.pattern_objects import ObservationExpression, \
+    ComparisonExpression, ComparisonComparators, Pattern, CombinedComparisonExpression, CombinedObservationExpression, \
+    ComparisonExpressionOperators
+
+logger = logger.set_logger(__name__)
+
+START_STOP_PATTERN = r"(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?Z)"
+STOP_TIME = datetime.utcnow()
+CONFIG_MAP_PATH = "json/config_map.json"
+
+# API query limit is 1350 (MAX_QUERY_LENGTH 1250 +TIMESTAMP_LENGTH 100)
+MAX_QUERY_LENGTH = 1250
+
+
+class FileNotFoundException(Exception):
+    pass
+
+
+class QueryStringPatternTranslator:
+    """
+    comparator values to match with supported data source operators
+    """
+
+    def __init__(self, pattern: Pattern, data_model_mapper, options):
+
+        logger.info("Nozomi Connector")
+        self.dmm = data_model_mapper
+        self.comparator_lookup = self.dmm.map_comparator()
+        self.config_map = self.load_json(CONFIG_MAP_PATH)
+        self.pattern = pattern
+        self.options = options
+        self.qualified_queries = []
+        self.logged = False
+        self.parse_expression(pattern)
+
+    @staticmethod
+    def load_json(rel_path_of_file) -> dict:
+        """ Consumes a json file and returns a dictionary
+        :param rel_path_of_file: str
+        :return: dict """
+        _json_path = path.dirname(path.realpath(__file__)) + "/" + rel_path_of_file
+        try:
+            if path.exists(_json_path):
+                with open(_json_path, encoding='utf-8') as f_obj:
+                    return json.load(f_obj)
+            raise FileNotFoundException
+        except FileNotFoundException as e:
+            raise FileNotFoundError(f'{rel_path_of_file} not found') from e
+
+    def _format_set(self, values, mapped_field_type, expression, mapped_fields_array) -> str:
+        """
+        Formats value in the event of set operation
+        :param values: list
+        :param mapped_field_type: str
+        :param expression: object
+        :param mapped_fields_array: object
+        :return formatted value
+        """
+        gen = values.element_iterator()
```

Comment:
What is "gen" in this case? This could be renamed to something more obvious.

Reply:
This name convention is followed from Synchronuos_template connector.